### PR TITLE
Labels for Account Settings

### PR DESCRIPTION
**Pull request by rszwajko**
Labels for following sections:

1. Console options - planned layout is section per console type: VNC, Spice, RDP, Serial
2. Resetting to default Account Settings
3. Server persistence opt-out feature - choosing the settings storage type (local or server-side)

### CODE REVIEW

**Comment by ovirt-infra** [comment]
Hello contributor, thanks for submitting a PR for this project! 

I am the bot who triggers "standard-CI" builds for this project.
As a security measure, I will not run automated tests on PRs that are not from white-listed contributors.

In order to allow automated tests to run, please ask one of the project maintainers to review the code and then do one of the following:

1. Type `ci test please` on this PR to trigger automated tests for it.
2. Type `ci add to whitelist` on this PR to trigger automated tests for it and also add you to the contributor white-list so that your future PRs will be tested automatically. ( keep in mind this list might be overwritten if the job XML is refreshed, for permanent whitelisting, please follow #3 option )
3. If you are planning to contribute to more than one project, maybe it's better to ask them to add you to the project organization, so you'll be able to run tests for all the organization's projects.

**Review by bamsalem — APPROVED** [review]
Since it's only strings and no actual code to test, without test it locally, LGTM

**Comment by sgratch** [comment]
ci add to whitelist

File: src/intl/messages.js
Comment (sgratch) [inline src/intl/messages.js:414]: What is this?
Is it referring to https://github.com/oVirt/ovirt-web-ui/issues/1393? I don't think/sure we need to enable this for all settings options. We discussed that only in the context of the locales. 
WDYT?

File: src/intl/messages.js
Comment (sgratch) [inline src/intl/messages.js:524]: "The chosen console **type** will..."

File: src/intl/messages.js
Comment (sgratch) [inline src/intl/messages.js:523]: I would change it to "console type"...

File: src/intl/messages.js
Comment (sgratch) [inline src/intl/messages.js:588]: Serial console settings is not needed on web-ui

File: src/intl/messages.js
Comment (sgratch) [inline src/intl/messages.js:590]: This should be considered in the context of locale settings only

File: src/intl/messages.js
Comment (sgratch) [inline src/intl/messages.js:595]: Why we need this?

File: src/intl/messages.js
Comment (sgratch) [inline src/intl/messages.js:617]: This is not required/planned as part of console options

File: src/intl/messages.js
Comment (sgratch) [inline src/intl/messages.js:729]: This is not required/planned as part of console options

File: src/intl/messages.js
Comment (sgratch) [inline src/intl/messages.js:732]: This is not required/planned as part of console options

File: src/intl/messages.js
Comment (sgratch) [inline src/intl/messages.js:563]: I would add:
"default settings (**installation defaults**)"
since those defaults are the hardcoded ones...

**Review by sgratch — CHANGES_REQUESTED** [review]
Few comments:

- "Automatic console connection" "automatically connect to the console..." - is missing
- account settings tabs (general, console, notifications...) are missing but we're still not sure about those so let's leave it out for now. The only Ttabs we can be sure of are maybe "general" and "console" so maybe adding only those (?)
- Please consider handling https://bugzilla.redhat.com/show_bug.cgi?id=1950323 on the way..

**Comment by rszwajko** [comment]
>   "Automatic console connection" "automatically connect to the console..." - is missing
>

Yes. We have already `connectAutomatically` label which seems to be used before. Auto-connect is also a VM specific settings (not part of the Account Settings).
 
>   account settings tabs (general, console, notifications...) are missing but we're still not sure about those so let's leave it out for now. The only Ttabs we can be sure of are maybe "general" and "console" so maybe adding only those (?)
> 

We have already labels for: General, Console, Notifications and Advanced options

>   Please consider handling https://bugzilla.redhat.com/show_bug.cgi?id=1950323 on the way..
>

This will require gerrit patch as it's Web Admin specific.

File: src/intl/messages.js
Comment (rszwajko) [reply src/intl/messages.js:414]: > Is it referring to #1393?

Yep. It will be (for now) only for language but my plan is to create  Advanced Options section and put it there.
My reasons:
1. adding this switch/checkbox next to language dropdown looks awkward
2. we are adding more options that require opt-out i.e. console options and notifications. Adding a switch/checkbox for every property would look bad.

File: src/intl/messages.js
Comment (rszwajko) [reply src/intl/messages.js:523]: Providing a correct naming gets tricky here... just "console type" could confuse users as this is not a console that will be used for **every** VM. It will be the first choice of course but console needs to be supported by the VM.

File: src/intl/messages.js
Comment (rszwajko) [reply src/intl/messages.js:588]: AFAIK SSH key is for serial console. The original label was a bit more descriptive - now it's reflected in the tooltip only. Originally it was planned in Console section.

File: src/intl/messages.js
Comment (rszwajko) [reply src/intl/messages.js:590]: Just wanted to cover all cases here. We have actually 3 options:
1. server-side - persisted as part of user profile i.e. refresh interval
2. client side - persisted in local storage i.e. locale
3. in-memory - not persisted at all - i.e. notifications

File: src/intl/messages.js
Comment (rszwajko) [reply src/intl/messages.js:595]: I plan to use it as section under Advanced Options to allow opt-out (currently only language).

File: src/intl/messages.js
Comment (rszwajko) [reply src/intl/messages.js:617]: 1. SPICE Options is a label for a section that groups all SPICE options (there are few of them)
2. SPICE Proxy is just in case here - it's present in the Web Admin UI

File: src/intl/messages.js
Comment (rszwajko) [reply src/intl/messages.js:729]: Added just in case to be compliant with Web Admin console options.

File: src/intl/messages.js
Comment (rszwajko) [reply src/intl/messages.js:732]: Added just in case to be compliant with Web Admin console options.

File: src/intl/messages.js
Comment (rszwajko) [reply src/intl/messages.js:563]: > default settings (installation defaults)
>

IMHO `installation defaults` sounds more complex then simple `default settings`. I would stick to the original design here.

**Comment by rszwajko** [comment]
1. Added `(Default)` label  - to be used in Select Box to mark the default options.
2. corrected `console -> console type`

File: src/intl/messages.js
Comment (sgratch) [reply src/intl/messages.js:527]: Right, but "Preferred Console" is also tricky from the same reason. That's why we might need a tooltip for explaining it.

File: src/intl/messages.js
Comment (sgratch) [reply src/intl/messages.js:418]: I'm not sure that opt-out for console options and notification are really needed. I can't think of a reason for this but ok.

File: src/intl/messages.js
Comment (sgratch) [reply src/intl/messages.js:592]: I prefer the current design (which fits the current implementation).
It can be in the console section.

File: src/intl/messages.js
Comment (sgratch) [reply src/intl/messages.js:594]: Let's have a separate discussion regarding this. It think that it might confuse users and not really needed to have those options. It might also complex our solution. 
For most settings - option 1 should be enabled unless we didn't implement this yet (notifications) or unless we have gaps that might break existing functionality like the locale.
option 2 is an internal solution since we also persisted the locale on server as well.

File: src/intl/messages.js
Comment (sgratch) [reply src/intl/messages.js:599]: I prefer a different label, more user friendly, like "persist account settings". This will require a tooltip as well....

File: src/intl/messages.js
Comment (sgratch) [reply src/intl/messages.js:621]: Not all options on webadmin are relevant for web-ui.
It was already decided about those that are relevant as part of Laura's design, based on older user portal version and customer requirements.
I prefer not to add settings that we are not planning to implement :)

File: src/intl/messages.js
Comment (sgratch) [reply src/intl/messages.js:733]: Same here

File: src/intl/messages.js
Comment (sgratch) [reply src/intl/messages.js:736]: Same here

File: src/intl/messages.js
Comment (sgratch) [reply src/intl/messages.js:567]: > > "Automatic console connection" "automatically connect to the console..." - is missing
> 
> Yes. We have already `connectAutomatically` label which seems to be used before. Auto-connect is also a VM specific settings (not part of the Account Settings).
> 

let's change to the labels suggested by Laura.


> > account settings tabs (general, console, notifications...) are missing but we're still not sure about those so let's leave it out for now. The only Ttabs we can be sure of are maybe "general" and "console" so maybe adding only those (?)
> 
> We have already labels for: General, Console, Notifications and Advanced options
> 

+1

> > Please consider handling https://bugzilla.redhat.com/show_bug.cgi?id=1950323 on the way..
> 
> This will require gerrit patch as it's Web Admin specific.

Right, my mistake.

**Comment by sgratch** [comment]
@rszwajko 
We might also add a tooltip (as appearing under each settings tab in origibal design) of:
image

**Comment by rszwajko** [comment]
ci please test

**Comment by rszwajko** [comment]
ci test please

File: src/intl/messages.js
Comment (rszwajko) [reply src/intl/messages.js:736]: technically we support it (but it's disabled):
https://github.com/oVirt/ovirt-web-ui/blob/53f33a8dc918b23e0f63498c4049fc70f9c794ab/src/sagas/console/rdpBuilder.js#L155

File: src/intl/messages.js
Comment (rszwajko) [reply src/intl/messages.js:592]: We might end up with leaving the current design but I would like to have some plan B here. Can we include them for now and decide later?

File: src/intl/messages.js
Comment (rszwajko) [reply src/intl/messages.js:599]: How about Advanced Options -> Storage -> Persist language +  checkbox or  switch?

Comment (sgratch) [reply src/intl/messages.js]: I don't like the "storage" label name. It's confusing with the storage settings for a VM.

**Review by sgratch — APPROVED** [review]
Overall it LGTM. If needed then we'll fix that later on.
- "Automatic console connection" - is missing. I prefer this suggested by Laura over the old connectAutomatically label
- "Enter serial console ssh key" -  is missing. I prefer this suggested by Laura over the old "ssh key" label, regardless to the section is resides in
- Few of the added labels are not surely needed.

**Pull request merged**
